internal/v2/fakes2av2/fakes2av2.go

```diff
@@ -66,6 +77,59 @@ func (s *Server) SetUpSession(stream s2av2pb.S2AService_SetUpSessionServer) erro
 	}
 }
 
+func offloadPrivateKeyOperation(req *s2av2pb.OffloadPrivateKeyOperationReq, hostname string) (*s2av2pb.SessionResp, error) {
+	switch x := req.GetOperation(); x{
+	case s2av2pb.OffloadPrivateKeyOperationReq_SIGN:
+		var root tls.Certificate
+		var err error
+		// Retrieve S2Av2 implementation of crypto.Signer.
+		switch hostname {
+		case "client_side_connection":
+			root, err = tls.X509KeyPair(clientRootCert, clientRootKey)
+			if err != nil {
+				return nil, err
+			}
+		case "server_side_connection":
+			root, err = tls.X509KeyPair(serverRootCert, serverRootKey)
+			if err != nil {
+				return nil, err
+			}
+		default:
+			s := fmt.Sprintf("Invalid hostname tied to SessionReq: %s", hostname)
+			return nil, errors.New(s)
+		}
+
+		// TODO(rmehta19): Parse req.GetSignatureAlgorithm to find Hash function that
+		// should be used here. Currently hardcoded to use SHA256.
+
+		// Generate the SHA256 hash of the block.
+		hsha256 := sha256.Sum256(req.GetInBytes())
```

Comment:
Why are these 2 lines needed?

Reply:
Line 106 is to generate the hash, and 107 is the convert from [32]byte to []byte, I can probably get rid of 107, and directly pass hsha256[:] to the Sign function.

Reply:
Sorry just stepping back: why do we need to hash the in bytes before passing them into the `Sign` API? That's unexpected to me. Do we need inputs to the `Sign` API to have e.g. an integer multiple of the hash digest length?

Reply:
Discussed Offline: S2A should just be signing whatever client populates in InBytes. Therefore, if we want to compute the hash of the InBytes, we should do that on the client side